Formula/pyside.rb

```diff
@@ -19,40 +19,73 @@ class Pyside < Formula
 
   depends_on "cmake" => :build
   depends_on "ninja" => :build
+  depends_on xcode: :build
   depends_on "llvm"
   depends_on "python@3.9"
   depends_on "qt"
 
+  uses_from_macos "libxml2"
+  uses_from_macos "libxslt"
+
   def install
-    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
+    inreplace "sources/pyside6/cmake/Macros/PySideModules.cmake",
+              "${shiboken_include_dirs}",
+              "${shiboken_include_dirs}:#{Formula["qt"].opt_include}"
 
-    args = std_cmake_args + %W[
-      -GNinja
-      -DPYTHON_EXECUTABLE=#{Formula["python@3.9"].opt_bin}/python#{xy}
-      -DCMAKE_INSTALL_RPATH=#{lib}
+    py = Formula["python@3.9"]
+    qt = Formula["qt"]
+    xy = Language::Python.major_minor_version py.opt_bin/"python3"
+    pyver = xy.to_s.delete "."
+    site_packages = prefix/Language::Python.site_packages("python3")
+    site_pyside = site_packages/"PySide6"
+    pyside_args = %W[
+      --no-examples
+      --no-qt-tools
+      --rpath #{opt_lib}
+      --shorter-paths
+      --skip-docs
     ]
+    system "python3", *Language::Python.setup_install_args(prefix), *pyside_args
 
-    mkdir "build" do
-      system "cmake", *args, ".."
-      system "ninja", "install"
+    # match cmake directory layout
+    %w[cmake pkgconfig].each { |d| lib.install "pyside3_install/p#{xy}/lib/#{d}" }
+    %w[PySide6 shiboken6].each { |l| lib.install Dir[site_packages/l/"lib*"] }
+    %w[pyside6 shiboken6].each do |l|
+      n = "lib#{l}.cpython-#{pyver}-darwin"
+      lib.install_symlink lib/shared_library(n, qt.version.major_minor) => shared_library(n)
+      inreplace lib/"pkgconfig/#{l}.pc",
+                "#{buildpath}/pyside3_install/p#{xy}".delete_prefix("/private"),
+                opt_prefix.to_s
     end
+    MachO::Tools.change_rpath "#{site_packages}/shiboken6/Shiboken.cpython-#{pyver}-darwin.so",
+      "#{buildpath}/pyside3_build/p#{xy}/shiboken6/libshiboken".delete_prefix("/private"), opt_lib.to_s
```

Comment:
This looks wrong. Why does it still have the build prefix as the RPATH?

Reply:
Because here uses the `setup.py` script, from official README, [cmake super project](https://code.qt.io/cgit/pyside/pyside-setup.git/tree/README.md#n207) is for development convenience. `setup.py` sets `CMAKE_INSTALL_PREFIX` to `buildpath/pyside3_build/p3.9`, so does the `rpath`, official [shiboken](https://pypi.org/project/shiboken6/#files) wheel also sets the rpath to the build path. In order to keep the directory layout same as previous `pyside` version, which is the test case expect , we need manually move these files...

Reply:
If upstream have changed the directory layout then we should probably just follow them.